components/Details.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+
+import { transformDetailsToFields } from './../utils/transformData';
+import { FormInput } from './Form/FormInput';
+export const Details = ({ detailsDataCallback, defaultValue }) => {
+  const [detailFields, setDetailFields] = useState(
+    transformDetailsToFields(defaultValue) || [],
+  );
+
+  const handleAddFields = (event) => {
+    event.preventDefault();
+    setDetailFields([...detailFields, { fieldName: '', fieldContent: '' }]);
+  };
+
+  const handleDeleteFields = (index, event) => {
+    event.preventDefault();
+    const data = [...detailFields];
+    data.splice(index, 1);
+    setDetailFields(data);
+  };
+
+  const handleFormChange = (index, event) => {
+    const data = [...detailFields];
+    data[index][event.target.name] = event.target.value;
+    setDetailFields(data);
+  };
+
+  useEffect(() => {
```

Comment:
Component should follow order of logical groups as follows

1. selectors/data/router
2. useState
3. useEffect
4. handler
5. dataTransformation for JSX